app/assets/javascripts/backbone/views/breakeven_product_total_view.coffee

```diff
@@ -0,0 +1,104 @@
+class MS.Views.BreakevenProductTotalView extends Backbone.View
+
+  # The view is called from the editiable table view.
+
+  # events:
+  #   'click tr [data-action="delete"]': 'removeRow'
+  #   'click .actions [data-action="add"]': 'addRow'
+
+  initialize: (options) ->
+    @products = options.products
+
+    @totalFixedCosts = parseFloat(@$('.total_fixed_costs').val())
+
+    _.each @products, (product) =>
+      product.on "product:changed", () =>
+        @calculateTotals()
+
+    # Defer loading of totals until the dom is ready
+    $ =>
+      @calculateTotals()
+
+  calculateTotals: ->
+    @notifyProducts()
+
+    totalPrice = @totalPrice()
+    totalCost = @totalCost()
+    totalQuantity = @totalQuantity()
+    totalRevenue = @totalRevenue()
+    totalPercentageOfSales = @totalPercentageOfSales()
+    totalProductCost = @totalProductCost()
+    Q = @Q()
+
+    totals = {totalPrice, totalCost, totalQuantity, totalRevenue, totalPercentageOfSales, totalProductCost, @totalFixedCosts, Q}
+    @writeTotalsToDom(totals)
+
+  totalPrice: ->
+    _.reduce(@products, (acc, product) =>
+      acc += product.price() if product.isValid()
+
+      return acc
+    , 0)
+
+  totalCost: ->
+    _.reduce(@products, (acc, product) =>
+      acc += product.cost() if product.isValid()
+
+      return acc
+    , 0)
+
+  totalProductCost: ->
+    _.reduce(@products, (acc, product) =>
+      acc += product.totalCost() if product.isValid()
+
+      return acc
+    , 0)
+
+  totalQuantity: ->
+    _.reduce(@products, (acc, product) =>
+      acc += product.quantity() if product.isValid()
+
+      return acc
+    , 0)
+
+  totalPs: ->
+    _.reduce(@products, (acc, product) =>
+      acc += product.ps() if product.isValid()
+
+      return acc
+    , 0.0)
+
+  totalRevenue: ->
+    _.reduce(@products, (acc, product) =>
+      acc += product.revenue() if product.isValid()
+
+      return acc
+    , 0.0)
+
+
+  totalPercentageOfSales: ->
+    _.reduce(@products, (acc, product) =>
+      acc += product.percentageOfSales() if product.isValid()
+
+      return acc
+    , 0)
+
+  Q: ->
+    @totalFixedCosts / @totalPs()
+
+  notifyProducts: () ->
+    _.each @products, (product) =>
+      product.totalsUpdated({@totalFixedCosts, Q: @Q()})
```

Comment:
I think because of the way the array is passed in. I could cast it to an array. `Array(@products).forEach` Would this be preferrable over underscore?